packages/discord.js/src/structures/Presence.js

```diff
@@ -389,8 +391,24 @@ class RichPresenceAssets {
 
     return this.activity.presence.client.rest.cdn.appAsset(this.activity.applicationId, this.largeImage, options);
   }
+
+  /**
+   * Tests whether an image key is a URL or not
+   * @param {string} imageKey The image key to test
+   * @returns {boolean}
+   * @private
+   */
+  static _isURLImage(imageKey) {
+    return RichPresenceAssets.URL_PATTERN.test(imageKey);
+  }
 }
 
+/**
+ * Regular expression that matches images URLs from image keys
+ * @type {RegExp}
+ */
+RichPresenceAssets.URL_PATTERN = /(https?|ftp|file):\/\//gm;
```

Comment:
```suggestion
RichPresenceAssets.URL_PATTERN = /^(?:https?|ftp|file):\/\//gm;
```

Reply:
`gm` flags sound redundant in this scenario

Reply:
The URLs also don't start with https/ftp/file but rather with mp://some_thing/url according to the shown videos in DDevs